### PR TITLE
Add proto compilation script

### PR DESCRIPTION
**Pull request by landrito**
@pongad can you take a look?

### CODE REVIEW

File: cmd/compile_protos/main.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	// Check if protoc is installed.
+	cmd := exec.Command("protoc", "--version")
```
Comment (pongad) [inline cmd/compile_protos/main.go]: `exec.Command(...).Run()` is perfectly OK to chain.

File: cmd/compile_protos/main.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	if err := cmd.Run(); err != nil {
+		log.Fatal("Error: 'protoc' is expected to be installed on the path.")
+	}
+
+	// Get proto dependencies
+	gopath := os.Getenv("GOPATH")
+	showcaseDir := gopath + "/" + "src/github.com/googleapis/gapic-showcase"
```
Comment (pongad) [inline cmd/compile_protos/main.go]: Here and below: Prefer `filepath.Join` here in case we run on Windows?

File: cmd/compile_protos/main.go (hunk 1 above)
Comment (landrito) [reply cmd/compile_protos/main.go]: Done.

File: cmd/compile_protos/main.go [hunk 3]
```diff
@@ -58,8 +57,7 @@ func main() {
 		"google",
 		"showcase",
 		"v1alpha2")
-	err = exec.Command("mkdir", "-p", protoDest).Run()
-	if err != nil {
+	if err = exec.Command("mkdir", "-p", protoDest).Run(); err != nil {
```
Comment (pongad) [inline cmd/compile_protos/main.go]: Up to you, but you can use `os.MkdirAll` here.

**Comment by landrito** [comment]
Thanks for the review @pongad!

**Pull request merged**